src/celeritas/em/distribution/UrbanMscStepLimit.hh

```diff
@@ -138,9 +138,6 @@ UrbanMscStepLimit::UrbanMscStepLimit(const UrbanMscRef&       shared,
     range_ = helper_.range();
     // Mean free path for MSC at current energy
     lambda_ = helper_.msc_mfp(inc_energy_);
-
-    // The slowing down range should already have been applied as a step limit
-    CELER_ENSURE(range_ >= phys_step_);
```

Comment:
I think this is the third time you've submitted a PR deleting this assertion 😂

I know you wrote on slack that these numbers are very slightly different, but this assertion is there to catch that difference so we can track it down! If it's because we're recalculating the range, we should figure out if caching will help eliminate the difference and improve performance, etc. If the assertion is getting in the way of further analysis, instead of deleting, loosen it a bit (`CELER_ENSURE(range_ >= phys_step_ || soft_equal(range_, phys_step_))`), add a comment, and make sure that nothing else in the class fails when this restriction is broken.

Reply:
Since I did not fully agree the assume that msc is invoked after the eloss range limitation, i.e, the current demo workflow, I still think that it is better to remove this hard restriction here- I mentioned a case for the msc process without the ionization (or bream) process itself, which is also a perfect fine application or test, but you want to ignore anything but the current workflow.  Anyway, we probably do not want to waste our time to argue about potential exceptions or general extensions for the user cases of msc.  Since you refactored this from the original code, I leave to introduce the better solution (probably the caching the range) after merging this MR wihtout any clutch from testem3 test by this assertion.